internal/syncd/syncd.go

```diff
@@ -0,0 +1,89 @@
+package syncd
+
+import (
+	"context"
+	"sync"
+
+	"ship-it/internal"
+
+	"k8s.io/helm/pkg/proto/hapi/chart"
+)
+
+// ImageReconciler is a callback that reconciles a new image with some external
+// state. For example, updating the values of a helm chart in a remote
+// repository to use the new image's tag.
+type ImageReconciler interface {
+	Reconcile(context.Context, *internal.Image) error
+}
+
+// ImageListener models a stream of `*Image` events. It calls the
```

Comment:
Math trivia: the claim that this interface "models a stream" can be formalized in a stronger type system, by applying a property of [CPS](https://en.wikipedia.org/wiki/Continuation-passing_style) functions over some monad. Practical category theory 🤓